### PR TITLE
Restricted Shared Service: API and UI Screens

### PR DESCRIPTION
**Pull request by damoodamoo**
# Resolves issue #2237 

- List endpoint for shared-services now returns a `RestrictedResource` list, which limits the properties that can be returned to `display_name`, `description` and 'overview`.
- Get by shared service ID endpoint now allows TRE users to call it (it was previously limited to TRE Admins only), and checks if the user is a TRE Admin or not. If an admin, returns the full fat object, else a restricted one. This was needed as a user may land on a restricted shared service page in the UI and need to load just that item.
- The resource views in the UI now have an optional `readonly` property, that can be passed in and prevent rendering of all options and menus, regardless of user role.
- Workspace services in the Workspace use the `readonly` property to make sure Admins in the workspace are not confused into thinking that the shared service relates to the workspace, or can be managed separately here.

Card view: 
image

Resource page showing shared service in the workspace:
image

### CODE REVIEW

File: api_app/models/domain/restricted_resource.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from models.domain.azuretremodel import AzureTREModel
+
+
+class RestrictedProperties(AzureTREModel):
+    display_name: str = ""
+    description: str = ""
+    overview: str = ""
+
+
+class RestrictedResource(AzureTREModel):
```
Comment (tanya-borisova) [inline api_app/models/domain/restricted_resource.py:13]: Out of curiousity, what were the fields in shared service that we want masked?

**Review by tanya-borisova — COMMENTED** [review]
I've only looked at Python part. Looks good, just have one question

File: api_app/models/domain/restricted_resource.py (hunk 1 above)
Comment (damoodamoo) [reply api_app/models/domain/restricted_resource.py:13]: anything in the properties object that's NOT name/description basically - which would include things like firewall rules / certificate names etc potentially

**Review by marrobi — APPROVED** [review]
Screen shot looks fine.

**Comment by damoodamoo** [comment]
/test-force-approve
Ran shared service tests previously

**Pull request merged**